### PR TITLE
Add Dynamic Loaded Items Spec

### PR DESCRIPTION
**Pull request by michaelwehrley**
Add Dynamic Loaded Items Spec

* Add js test that validates the dynamic load of shared items.
* Update TFD Engine to latest 3.2.31 - necessary for tests to pass.

[Finished: #97469646]

### CODE REVIEW

**Comment by ericcf** [comment]
:facepunch: go ahead and merge when it passes!

**Comment by cbrenner04** [comment]
:+1:

File: spec/features/participant/tool_spec.rb [hunk 1]
```diff
@@ -28,4 +28,24 @@
...
+      page.driver.resize(SCREEN_WIDTH, SCREEN_HEIGHT)
+      stub_const("SocialNetworking::SharedItemController::SHARED_ITEM_PAGE_SIZE", SHARED_ITEM_LIMIT)
+    end
+
+    it "Dynamically loads more shared content", :js do
+      visit "/"
+
+      expect(page).not_to have_text("it's always sunny in Philadelphia")
+
+      page.driver.scroll_to(0, SCREEN_HEIGHT)
```
Comment (cbrenner04) [inline spec/features/participant/tool_spec.rb]: @michaelwehrley you may need to scroll more than once depending upon how many items there are above this one

**Comment by cbrenner04** [comment]
@michaelwehrley I still think you need to add some logic to that scroll. Since you are using all fixtures, if more shared items are added to the fixtures, it is likely you will need to scroll more than once to find this shared item. As it is now this is a pretty brittle spec.

**Comment by ericcf** [comment]
Why not programmatically destroy the feed items and then create just the number you want within the spec? That way you guarantee that the number will not change.

**Comment by michaelwehrley** [comment]
I think this is one of the conundrums of fixtures.  The same thing happens to other specs when we add fixtures - the total count changes.  I will however think about other options.  Thanks!

**Comment by cbrenner04** [comment]
That's true which is why we have suggested when looking for a count in a spec to look for the change in the count not the total number

**Comment by michaelwehrley** [comment]
@ericcf The problem was not with either the number of fixtures nor not knowing what was there - I set the feed to 3 and I know the 6 items available.  It failed the first time because the call to the database took too long.

**Comment by ericcf** [comment]
I'm ok with this for now. If/when it breaks we'll have to address it again.

**Pull request merged**